docs/assembly.md

```diff
@@ -0,0 +1,184 @@
+        change on a *minor* version release)
+      + `stable` denotes an API that is safe to use in production systems and
+        are subject ot semantic versioning guarantees (they may not receive
+        breaking changes without a *major* version bump)
+      + `deprecated` denotes an API that should no longer be used. The
+        `deprecated` entry in the `docs` object should contain a message
+        explaining how users should migrate away
+      + `external` denotes an API that is not owned by the package's maintainer
+        and may change in unexpected ways. Such APIs are usually derived from
+        external artifacts, which the package maintainers do not have control
+        over.
+  * additional entries represent user-defined `JSDoc` tags with meaning defined
+    by convention and/or the package maintainer
+* `locationInModule` - coordinates of the declaration in the source
+  * `fileName` - the path to the source file, relative to the package root
+  * `line` - the line number on which the entity is declared (or the first line
+    when a declaration spans multiple lines)
+
+### Types
+
+#### Classes
+
+Attribute    | Type        | Description
+-------------|-------------|----------------------------------------------------
+`kind`       |`'class'`    |Discriminator to identify classes
+`abstract`   |`boolean`    |Whether this class is *abstract*
+`assembly`   |`string`     |The name of the assembly this class is a part of
+`base`       |`string`     |The fully-qualified name of the parent class of this class
+`fqn`        |`string`     |The fully-qualified name of the class
+`initializer`|`Constructor`|The class' [constructor]
+`interfaces` |`string[]`   |The fully-qualified names of interfaces implemented by this class
+`methods`    |`Method[]`   |The [methods] declared by this class
+`name`       |`string`     |The simple name of the class
+`properties` |`Property[]` |The [properties] declared by this class
+
+[constructor]: #constructors
+[interfaces]: #interfaces
+[methods]: #methods
+[properties]: #properties
+
+#### Interfaces
+
+`jsii` interfaces are declarations of type signatures that can be implemented by
+classes. *Interface* names must be prefixed with an `I` (e.g: `IFoo`).
+
+Attribute    | Type        | Description
+-------------|-------------|----------------------------------------------------
+`kind`       |`'inteface'` |Discriminator to identify interfaces
+`assembly`   |`string`     |The name of the assembly this interface is a part of
+`fqn`        |`string`     |The fully-qualified name of the interface
+`interfaces` |`string[]`   |The fully-qualified names of interfaces extended by this inteface
+`methods`    |`Method[]`   |The [methods] declared by this interface
+`name`       |`string`     |The simple name of the interface
+`properties` |`Property[]` |The [properties] declared by this interface
+
+#### Structures (a.k.a. Data Types)
```

Comment:
This should be `structs` not `structures`

Reply:
Yeah I tried to take a stance against abbreviating here... How strongly do you feel about that?  Like - do you feel the common use here is sufficient that "structs" conveys additional meaning than "structures"?

Reply:
Oh well never mind, I changed.